python/tvm/relay/op/transform.py

```diff
@@ -1087,6 +1087,10 @@ def gather_nd(data, indices, batch_dims=0):
     batch_dims : int
         The number of batch dimensions.
 
+    num_indices_per_tuple : int
+        The size of an indexing tuple, which is a fixed value and the same as indices.shape[0]
+        Only needed when other dimensions of indices are dynamic.
```

Comment:
does the default value of -1 have some significance or does it just mean its not used by default?

Reply:
It is not significant.  Having a default value at call sites means this value is not explicitly set, which is an error for dynamic workload. The negative value is only for catching an error at https://github.com/apache/tvm/blob/55a4430f507d801b7441e60d00f06c8161aea30a/python/tvm/relay/op/_transform.py#L1106. 

This value is not optional either, since this is required for dynamic workloads.

Reply:
Would it be inconvenient to use `None` instead of `-1`? That would be more consistent with the behavior in my opinion.

Reply:
Would it be inconvenient to make the default value `None` instead of `-1`? That would be more consistent with the behavior and less confusing.

Reply:
Ok I can make it an `tvm::Optional<Integer>`.

Reply:
Made it Optional with default None.